05-mvc-with-rails/rails-sessions-and-auth-with-devise/readme.md

```diff
@@ -0,0 +1,365 @@
+
+Here's some talking points:
+- What does it mean for a user to be signed in?
+- For that matter, what does it mean to be a user? How are we encapsulating that data.
+- We've used our backend to store data about apartments and tenants. Can we use it to store data about a user itself?
+
+> We need to start thinking about how to capture information into a User model. That has information about username, email, or password. We also need a way to transfer or perserve state in our applications from request to request.
+
+## Cookies
+
+### We do:
+1. In Chrome, go to Preferences
+- Show advanced settings
+- Click "Content settings"
+- Click "All cookies and site data"
+- Scroll to `github.com`
+
+Click around on the different "buttons" or "tabs". In particular, look at `dotcom_user`, `signed_in`, and `tz`.
+
+##### Turn and talk: What do you see? What do these do?
+
+A *cookie* is a piece of data stored on your computer by your web browser and associated with a particular website.
+
+### Example cookies:
+
+From Github:
+```
+Name: dotcom_user
+Content: RobertAKARobin
+Domain: .github.com
+Path: /
+Send for: Secure connections only
+Accessible to script: No (HttpOnly)
+Created: Wednesday, July 22, 2015 at 7:04:22 PM
+Expires: Sunday, July 22, 2035 at 7:04:22 PM
+```
+
+## Enter Sessions(i do, talk + setup)15m
+- HTTP is stateless, so it's up to either the browser or application that needs to "remember" what's happening
+- A session is just a place to store data during one request that you can read during later requests.
+- A session is a hash containing key value pairs that provide state in your application
+
+Let's hop into some code so we can see sessions in action.
+
+The code we'll be working with today can be found [here](https://github.com/ga-dc/tunr_rails_sessions_devise) This is a version of the tunr rails application we've been building on this week. It contains everything that we've learned up until now.
+
+Make sure you fork/clone this repo if you would like to follow along. It is not too important to follow along for this portion of this lesson. I encourage you to just watch and use the following code to practice it yourself later.
+
+## Set and Use session key-value pairs
+
+In order to understand sessions, we're going to add some code to our existing artists controller. In `app/controllers/artists_controller.rb`:
+
+```ruby
+# this action will set `test: "this string was set on the testing_session action"` as a key value pair in the session
+def testing_session
+  session[:test] = "this string was set on the testing_session action"
+end
+
+# also add @test = session[:test] to index action
+def index
+  @test = session[:test]
+  @artists = Artist.all
+end
```

Comment:
Would love for us to come up with a more practical and useful example, one that's just as simple as this. Let's brainstorm tomorrow as soon as possible. Maybe something like a sort order for artists.